### PR TITLE
move delete logic to a pkg

### PR DESCRIPTION
**Pull request by marekschwarz**
As in #4780 discussed, I moved the logic for the deletion of profiles and machines to the pkg folder, to decouple the command logic from the delete logic.
For this I introduced a new folder named `delete`, where I moved all the logic and testdata. I'm not happy with the name of the folder and created it because of import cycles, but we can discuss the name or the place where the logic sits if you want.
While doing this had to move the `getClusterBootstrapper`-Method from `root.go` to `cluster.go`.

### CODE REVIEW

**Comment by sharifelgamal** [comment]
@minikube-bot OK to test

**Comment by medyagh** [comment]
@marekschwarz  can you please provide a PR description of what it does and how it will affect other things?

**Comment by marekschwarz** [comment]
> @marekschwarz can you please provide a PR description of what it does and how it will affect other things?

Done

**Comment by medyagh** [comment]
while I didn't study the PR, but a quick look I noticed it doesn't Read well

for example `delete.RemoveProfiles()` doesn't read natural to me.
I don't know what would be a better name to suggest since I haven't studied it. but I believe the code should read nice and not stutter.

for example

action.DeleteAll()

**Comment by marekschwarz** [comment]
> while I didn't study the PR, but a quick look I noticed it doesn't Read well
> 
> for example `delete.RemoveProfiles()` doesn't read natural to me.
> I don't know what would be a better name to suggest since I haven't studied it. but I believe the code should read nice and not stutter.
> 
> for example
> `action.DeleteAll()`
> or
> `profile.DeleteAll()`

Thanks for the suggestion.
Didn't think a lot  about names at that point and just wanted to make it work and figure out a good name afterwards.
I named it `profile.DeleteAll()`, like you suggested.

I moved `KillMountProcess()` to `pkg/minikube/cluster/mount.go`.
Can you suggest a good place for `UninstallKubernetes()`? It doesn't seem right in `pkg/minikube/profile/delete.go`.

I can try to bring `profile.go` over to this pkg, so that we got all profile-related code together. May be better there than in `pkg/minikube/config`.

**Comment by medyagh** [comment]
hm...thats a good question, I am wondering if cluster pkg or bootstrapper... the question is who uses this function and what it depends on

**Comment by marekschwarz** [comment]
> hm...thats a good question, my quick judgement is maybe `cluster` pkg but I could easily be wrong ! the question is who uses this function and what it depends on

It is only used by `pkg/minikube/profile/delete.go`. Depends on `libmachine.API`, `config.KubernetesConfig` and `cluster.GetClusterBootstrapper()`.
I think somewhere in the cluster-pkg may be the right place.

**Comment by marekschwarz** [comment]
verify

File: cmd/minikube/cmd/logs.go [hunk 1]
```diff
@@ -66,7 +67,7 @@ var logsCmd = &cobra.Command{
 		if err != nil {
 			exit.WithError("command runner", err)
 		}
-		bs, err := getClusterBootstrapper(api, viper.GetString(cmdcfg.Bootstrapper))
+		bs, err := cluster.GetClusterBootstrapper(api, viper.GetString(cmdcfg.Bootstrapper))
```
Comment (medyagh) [inline cmd/minikube/cmd/logs.go]: this name stutters, I suggets, a better name like `cluseter.Bootstrapper(...`

File: pkg/minikube/cluster/mount.go [hunk 2]
```diff
@@ -145,3 +150,51 @@ func Unmount(r mountRunner, target string) error {
 	glog.Infof("unmount for %s ran successfully", target)
 	return nil
 }
+
+// killMountProcess kills the mount process, if it is running
+func KillMountProcess() error {
```
Comment (medyagh) [inline pkg/minikube/cluster/mount.go]: the name in the comment needs to be capital KillMount...

**Comment by medyagh** [comment]
@marekschwarz sorry for the delay can u plz rebase

**Comment by tstromberg** [comment]
Do you mind resolving the merge conflicts?

**Comment by marekschwarz** [comment]
> Do you mind resolving the merge conflicts?

I will take care of it this week :-)

**Comment by medyagh** [comment]
@marekschwarz please make sure to run make test  to see the test failures and make lint to see lint issues locally

**Comment by medyagh** [comment]
@marekschwarz are you still working on this ?

**Comment by medyagh** [comment]
@marekschwarz I close this PR for now, please feel free to re-open when ready

**Pull request closed without merging**